src/main/java/com/netflix/imflibrary/MXFPropertyPopulator.java

```diff
@@ -111,6 +112,10 @@ else if (field.getType() == CompoundDataTypes.Timestamp.class)
                 CompoundDataTypes.Timestamp timestamp = new CompoundDataTypes.Timestamp(byteProvider);
                 field.set(object, timestamp);
             }
+            else if (field.getType() == J2KExtendedCapabilities.class) {
+                J2KExtendedCapabilities j2KExtendedCapabilities = new J2KExtendedCapabilities(byteProvider);
+                field.set(object, j2KExtendedCapabilities);
+            }
```

Comment:
I agree that it does _feel_ very clean to recognize this as a top-level type. But I can't immediately think of a cleaner way to handle this within the current code. We are also _already_ doing something similar for `JPEG2000PictureComponentBO` a little further down. So this is probably okay.

We can consider this as something to clean up in a future refactoring, if we make improvements to the way we recognize MXF properties.